tests/utils.py

```diff
@@ -4,25 +4,42 @@
 import snap_http
 
 
-def call_and_await_api(name: str, *args: Any, **kwargs: Any) -> snap_http.SnapdResponse:
-    """Call the `name` endpoint and wait until changes are applied."""
-    f: Callable[..., snap_http.SnapdResponse] = getattr(snap_http, name)
-    response = f(*args, **kwargs)
+def wait_for(
+    func: Callable[..., snap_http.SnapdResponse]
+) -> Callable[..., snap_http.SnapdResponse]:
+    """Call `func` and wait for changes to be applied in snapd."""
 
-    if response.type == "sync":
-        return response
+    def wrapper(*args: Any, **kwargs: Any) -> snap_http.SnapdResponse:
+        response = func(*args, **kwargs)
 
-    change = response.change
-    while True:
-        time.sleep(0.1)
+        if response.type == "sync":
+            return response
 
-        status = snap_http.check_change(change).result
-        if status["status"] in snap_http.COMPLETE_STATUSES:
-            break
+        change = response.change
+        while True:
+            time.sleep(0.1)
 
-    return response
+            status = snap_http.check_change(change).result
+            if status["status"] in snap_http.COMPLETE_STATUSES:
+                return response
+
+    return wrapper
 
 
 def is_snap_installed(snap_name: str) -> bool:
     """Check if the snap with name `snap_name` is installed."""
     return snap_name in {snap["name"] for snap in snap_http.list().result}
+
+
+@wait_for
+def sideload_snap(file_path: str) -> bool:
```

Comment:
plans to beef up `http.install` with side-loading capability, or is an alternative call, (e.g., `http.sideload`) more apt?

Reply:
I was thinking of `http.sideload` since the function signatures differ a lot. I could work on that in a follow-up PR